src/graph_compute/io.cpp

```diff
@@ -94,3 +95,73 @@ int readEdgesFromFile(const string filepath,
 
   return edgelistfile_read(filepath, &builder);
 }
+
+void makeSimpleAndUndirected(vertex_t * nodes,
```

Comment:
That's correct, each of the scheduling algorithms assumes a simple, undirected graph.  I'm happy to remove this if we can credibly guarantee to provide only simple, undirected graphs.